scripts/validator_msig.sh

```diff
@@ -69,60 +53,65 @@ awk '{
-            echo "INFO: tonos-cli submitTransaction attempt #${i}..."
-            if ! "${UTILS_DIR}/tonos-cli" call "${MSIG_ADDR}" submitTransaction \
-                "{\"dest\":\"${elector_addr}\",\"value\":\"1000000000\",\"bounce\":true,\"allBalance\":false,\"payload\":\"${recover_query_boc}\"}" \
-                --abi "${CONFIGS_DIR}/SafeMultisigWallet.abi.json" \
-                --sign "${KEYS_DIR}/msig.keys.json"; then
-                echo "INFO: tonos-cli submitTransaction attempt #${i}... FAIL"
-            else
-                echo "INFO: tonos-cli submitTransaction attempt #${i}... PASS"
-                break
-            fi
-        done
+"${TON_BUILD_DIR}/lite-client/lite-client" \
+    -p "${KEYS_DIR}/liteserver.pub" -a 127.0.0.1:3031 \
+    -rc "runmethod ${elector_addr} compute_returned_stake 0x$(echo "${MSIG_ADDR}" | cut -d ':' -f 2)" \
+    -rc "quit" &>"${ELECTIONS_WORK_DIR}/recover-state"
 
-        date +"INFO: %F %T Recover of $recover_amount GR requested"
-    fi
+awk '{
+    if ($1 == "result:") {
+        print $3
+    }
+}' "${ELECTIONS_WORK_DIR}/recover-state" >"${ELECTIONS_WORK_DIR}/recover-amount"
+
+recover_amount=$(cat "${ELECTIONS_WORK_DIR}/recover-amount")
+echo "INFO: recover_amount = ${recover_amount} nanotokens"
+
+if [ "$recover_amount" != "0" ]; then
+    "${TON_BUILD_DIR}/crypto/fift" -I "${TON_SRC_DIR}/crypto/fift/lib:${TON_SRC_DIR}/crypto/smartcont" -s recover-stake.fif "${ELECTIONS_WORK_DIR}/recover-query.boc"
+
+    recover_query_boc=$(base64 --wrap=0 "${ELECTIONS_WORK_DIR}/recover-query.boc")
+
+    for i in $(seq ${TONOS_CLI_SEND_ATTEMPTS}); do
+        echo "INFO: tonos-cli submitTransaction attempt #${i}..."
+        set -x
+        if ! "${UTILS_DIR}/tonos-cli" call "${MSIG_ADDR}" submitTransaction \
+            "{\"dest\":\"${elector_addr}\",\"value\":\"1000000000\",\"bounce\":true,\"allBalance\":false,\"payload\":\"${recover_query_boc}\"}" \
+            --abi "${CONFIGS_DIR}/SafeMultisigWallet.abi.json" \
+            --sign "${KEYS_DIR}/msig.keys.json"; then
+            echo "INFO: tonos-cli submitTransaction attempt #${i}... FAIL"
+        else
+            echo "INFO: tonos-cli submitTransaction attempt #${i}... PASS"
+            break
+        fi
+        set +x
+    done
+
+    date +"INFO: %F %T Recover of $recover_amount GR requested"
+    exit 0
+else
+    echo "INFO: nothing to recover"
+fi
 
+if [ "$election_id" == "0" ]; then
+    date +"INFO: %F %T No current elections"
     echo "INFO: $(basename "$0") END $(date +%s) / $(date)"
-    exit
+    exit 0
 fi
 
 if [ -f "${ELECTIONS_WORK_DIR}/stop-election" ]; then
     echo "INFO: $(basename "$0") END $(date +%s) / $(date)"
-    exit
+    exit 0
 fi
 
 if [ -f "${ELECTIONS_WORK_DIR}/active-election-id" ]; then
     active_election_id=$(cat "${ELECTIONS_WORK_DIR}/active-election-id")
     if [ "$active_election_id" = "$election_id" ]; then
         date +"INFO: %F %T Elections $election_id, already submitted"
```

Comment:
The Elector contract allows additional stakes with the same validator key until the elections end. Why the script prevents the eleciton participants from sending more stakes during elections? I saw many situations in the community chats, when for some reason the first stake was unsuccessfull (e.g. incorrect amount), and the script didn't allow to repeat sending stakes to the same elections, and they have to miss one round without no valid reason.

Reply:
Will be addressed as a separate PR.